tools/python_api/src_cpp/cached_import/py_cached_import.cpp

```diff
@@ -0,0 +1,18 @@
+#include "cached_import/py_cached_import.h"
+
+namespace kuzu {
+
+PythonCachedImport::~PythonCachedImport() {
+ 	py::gil_scoped_acquire acquire;
+	allObjects.clear();
+}
+
+py::handle PythonCachedImport::addToCache(py::object obj) {
+    auto ptr = obj.ptr();
+    allObjects.push_back(obj);
```

Comment:
This invokes a copy.

Reply:
This is only called from functions that have the GIL. Maybe we could make the function private?